packages/flutter/lib/src/material/dialog.dart

```diff
@@ -1564,30 +1554,65 @@ class DialogRoute<T> extends RawDialogRoute<T> {
   /// A dialog route with Material entrance and exit animations,
   /// modal barrier color, and modal barrier behavior (dialog is dismissible
   /// with a tap on the barrier).
-  DialogRoute({
+ factory DialogRoute({
     required BuildContext context,
     required WidgetBuilder builder,
     CapturedThemes? themes,
-    super.barrierColor = Colors.black54,
-    super.barrierDismissible,
+    Color? barrierColor = Colors.black54,
+    bool barrierDismissible = true,
     String? barrierLabel,
     bool useSafeArea = true,
-    super.settings,
-    super.anchorPoint,
-    super.traversalEdgeBehavior,
-  }) : super(
-         pageBuilder: (BuildContext buildContext, Animation<double> animation, Animation<double> secondaryAnimation) {
-           final Widget pageChild = Builder(builder: builder);
+    RouteSettings? settings,
+    Offset? anchorPoint,
+    TraversalEdgeBehavior? traversalEdgeBehavior,
+  }) {
+    final _DialogRouteAnimation animation = _DialogRouteAnimation();
+    return DialogRoute<T>._(animation,
+    pageBuilder: (BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
+       final Widget pageChild = Builder(builder: builder);
            Widget dialog = themes?.wrap(pageChild) ?? pageChild;
            if (useSafeArea) {
              dialog = SafeArea(child: dialog);
            }
            return dialog;
-         },
-         barrierLabel: barrierLabel ?? MaterialLocalizations.of(context).modalBarrierDismissLabel,
-         transitionDuration: const Duration(milliseconds: 150),
-         transitionBuilder: _buildMaterialDialogTransitions,
-       );
+    },
+    barrierLabel: barrierLabel ?? MaterialLocalizations.of(context).modalBarrierDismissLabel,
+    transitionDuration: const Duration(milliseconds: 150),
+    transitionBuilder: animation.buildMaterialDialogTransitions,
+    barrierColor: barrierColor,
+    barrierDismissible: barrierDismissible,
+    settings: settings,
+    anchorPoint: anchorPoint,
+    traversalEdgeBehavior: traversalEdgeBehavior,
+    );
+  }
+
+  DialogRoute._(this._animation, {required super.pageBuilder, super.barrierLabel, super.transitionDuration,  super.transitionBuilder, super.barrierColor, super.anchorPoint, super.barrierDismissible, super.settings, super.traversalEdgeBehavior});
+
+  _DialogRouteAnimation _animation;
+
+  @override
+  void dispose() {
+    _animation.dispose();
+    super.dispose();
+  }
+}
+/// Holds [DialogRoute] animation curves.
+class _DialogRouteAnimation {
```

Comment:
Can you move this class to the place of the method? It will make code review easier.